llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1340,7 +1340,8 @@ Value *HWAddressSanitizer::getFrameRecordInfo(IRBuilder<> &IRB) {
 }
 
 void HWAddressSanitizer::emitPrologue(IRBuilder<> &IRB, bool WithFrameRecord) {
-  if (!Mapping.InTls)
+  if (!Mapping.InTls ||
+      (Mapping.Offset != 0 && Mapping.Offset != kDynamicShadowSentinel))
```

Comment:
this is essentially `isFixedShadow`, right? maybe pull this into a variable of that name for documentation.

Reply:
https://github.com/llvm/llvm-project/pull/109616